File: app/src/main/java/ch/protonmail/android/utils/crypto/TextVerificationResult.java

package ch.protonmail.android.utils.crypto;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.Objects;


public class TextVerificationResult extends AbstractDecryptionResult {

    private final String data;
    private final Long verifiedSignatureTimestamp;

    public TextVerificationResult(@NonNull String data, boolean signatureIsValid, @Nullable Long verifiedSignatureTimestamp) {
        super(true, signatureIsValid);
        this.data = data;

        if (signatureIsValid) {
            this.verifiedSignatureTimestamp = Objects.requireNonNull(verifiedSignatureTimestamp, "verifiedSignatureTimestamp must not be null for valid signatures");
        } else if (verifiedSignatureTimestamp != null) {
            throw new IllegalArgumentException("verifiedSignatureTimestamp must be null for invalid signatures");
        } else {
            this.verifiedSignatureTimestamp = null;
        }
    }

    /**
     * @return creation time of the verified signature, or null for invalid signatures
     */
    @Nullable
    public Long getVerifiedSignatureTimestamp() {
        return verifiedSignatureTimestamp;
    }

    /**
     * @return signed data. Call `isVerified` to confirm signature validity.
     */
    public String getData() {
        return data;
    }
}

File: app/src/main/java/ch/protonmail/android/utils/crypto/TextVerificationResult.kt
/*
 * Copyright (c) 2022 Proton Technologies AG
 *
 * This file is part of ProtonMail.
 *
 * ProtonMail is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * ProtonMail is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with ProtonMail. If not, see https://www.gnu.org/licenses/.
 */

package ch.protonmail.android.utils.crypto

data class TextVerificationResult(
    /**
     * @return signed data. Call `isVerified` to confirm signature validity.
     */
    val data: String,
    val signatureIsValid: Boolean,
    val verifiedSignatureTimestamp: Long?,
): AbstractDecryptionResult(true, signatureIsValid) {
    init {
        if(signatureIsValid){
            requireNotNull(verifiedSignatureTimestamp){
                "verifiedSignatureTimestamp must not be null for valid signatures"
            }
        }else{
            require(verifiedSignatureTimestamp == null){
                "verifiedSignatureTimestamp must be null for invalid signatures"
            }
        }
    }
}